build.gradle.kts

```diff
@@ -50,22 +50,30 @@ dependencies {
     include(modImplementation("org.jetbrains.kotlinx:kotlinx-serialization-json:1.5.1")!!)
     include(modImplementation("com.github.char:Koffee:3a78d8a437")!!)
 
-    include(implementation(annotationProcessor("com.github.llamalad7.mixinextras:mixinextras-fabric:0.2.0-beta.8")!!)!!)
+    include(implementation(annotationProcessor("com.github.llamalad7.mixinextras:mixinextras-fabric:0.2.0")!!)!!)
 
     // 1.18 versions are good enough for Elementa and Vigilance, but not UC, so we
     // need to exclude this version
     configurations.modApi { exclude("gg.essential", "universalcraft-1.18.1-fabric") }
 
     include(modImplementation("gg.essential:vigilance-1.18.1-fabric:295")!!)
-    include(modImplementation("gg.essential:universalcraft-$platform:316")!!)
+    if (mcVersion == "1.20.4-fabric") {
+        include(modImplementation("gg.essential:universalcraft-1.20.3-fabric:319+feature-1.20.3-fabric")!!)
+    } else {
+        include(modImplementation("gg.essential:universalcraft-$platform:316")!!)
+    }
     include(modImplementation("gg.essential:elementa-1.18.1-fabric:610")!!)
 
-    val modMenuVersion = when (mcVersion) {
-        "1.20.1-fabric", "1.20.2-fabric" -> "7.2.1"
-        "1.19.4-fabric" -> "6.3.1"
-        else -> throw Exception("Minecraft version $modVersion is not supported")
+    let {
+        val modMenuVersion = when (mcVersion) {
+            // TODO: ModMenu for 1.20.4
```

Comment:
Modmenu 9.0 pre1 is available: https://modrinth.com/mod/modmenu/version/9.0.0-pre.1